tests/core/middleware/test_time_based_cache_middleware.py

```diff
@@ -1,125 +1,150 @@
 
-    middleware = construct_time_based_cache_middleware(
+    w3.middleware_stack.add(construct_time_based_cache_middleware(
         cache_class=cache_class,
         cache_expire_seconds=10,
         rpc_whitelist={'fake_endpoint'},
-    )(None, None)
+    ))
 
-    assert middleware('fake_endpoint', [1]) == 'value-a'
+    assert w3.manager.request_blocking('fake_endpoint', [1]) == 'value-a'
 
 
-def test_time_based_cache_middleware_populates_cache():
-    def make_request(method, params):
-        return {
-            'result': str(uuid.uuid4()),
-        }
+def test_time_based_cache_middleware_populates_cache(w3):
+    result = w3.manager.request_blocking('fake_endpoint', [])
 
-    middleware = construct_time_based_cache_middleware(
-        cache_class=dict,
-        cache_expire_seconds=10,
-        rpc_whitelist={'fake_endpoint'},
-    )(make_request, None)
-
-    response = middleware('fake_endpoint', [])
-    assert 'result' in response
-
-    assert middleware('fake_endpoint', []) == response
-    assert not middleware('fake_endpoint', [1]) == response
+    assert w3.manager.request_blocking('fake_endpoint', []) == result
+    assert not w3.manager.request_blocking('fake_endpoint', [1]) == result
 
 
-def test_time_based_cache_middleware_expires_old_values():
-    def make_request(method, params):
-        return {
-            'result': str(uuid.uuid4())
-        }
+def test_time_based_cache_middleware_expires_old_values(w3_base, result_generator_middleware):
+    w3 = w3_base
+    w3.middleware_stack.add(result_generator_middleware)
 
     def cache_class():
         return {
             generate_cache_key(('fake_endpoint', [1])): (
                 time.time() - 10,
-                'value-a',
+                {'result': 'value-a'},
             ),
         }
 
-    middleware = construct_time_based_cache_middleware(
+    w3.middleware_stack.add(construct_time_based_cache_middleware(
         cache_class=cache_class,
         cache_expire_seconds=10,
         rpc_whitelist={'fake_endpoint'},
-    )(make_request, None)
+    ))
 
-    response = middleware('fake_endpoint', [1])
-    assert not response == 'value-a'
-    assert 'result' in response
-
-    assert middleware('fake_endpoint', [1]) == response
+    result = w3.manager.request_blocking('fake_endpoint', [1])
+    assert not result == 'value-a'
+    assert w3.manager.request_blocking('fake_endpoint', [1]) == result
 
 
 @pytest.mark.parametrize(
     'response',
     (
         {},
         {'result': None},
-        {'error': 'some error message'},
     )
 )
-def test_time_based_cache_middleware_does_not_cache_bad_responses(response):
-    def make_request(method, params):
-        return assoc(response, 'id', str(uuid.uuid4()))
+def test_time_based_cache_middleware_does_not_cache_bad_responses(w3_base,
+                                                                  response,
+                                                                  time_cache_middleware):
```

Comment:
*style nitpick* guide says w3_base belongs on its own line with one indent

Reply:
This may take a few more reminders for me to break this habit.